(Publish TBD) Stripe Integration documentation (#14434)
newrelic/docs-website: src/content/docs/infrastructure/host-integrations/host-integrations-list/stripe-integration.mdx

---
title: Stripe integration
tags:
  - Stripe integration
  - New Relic integration
metaDescription: Install our Stripe dashboards and see your Stripe data in New Relic.  
---
import infrastructureStripeDashboard from 'images/infrastructure_screenshot-stripe-dashboard.webp'

> By utilizing New Relic Stripe dashboard, you gain the ability to monitor your application's transaction overview, transaction volume, revenue and transaction errors effortlessly. The Stripe dashboard is developed with our application specific monitoring agent, browser monitoring agent offering you a consolidated view of critical data, conveniently accessible in one central location.

Reviewer: ```suggestion
The New Relic Stripe dashboard provides a consolidated view of your application's transaction overview, transaction volume, revenue, and transaction errors. The dashboard is developed with our application-specific monitoring agent and browser monitoring agent, so you can easily monitor your application's performance in one central location.
```

<img
    title="Stripe"
    alt="Stripe dashboard"
    src={infrastructureStripeDashboard}
/>

<figcaption>

> After setting up our Stripe, you can install a dashboard for your Stripe metrics. 

Reviewer: ```suggestion
After setting up monitoring for your Stripe app, you can view all your Stripe metrics in one dashboard.
```

          "stripe.currency" : sessionJSON.currency,
          "stripe.payment_status": sessionJSON.payment_status,
          "stripe.status": sessionJSON.status,
          "stripe.discount_amount":sessionJSON.total_details.amount_discount,
          "stripe.shipping_amount":sessionJSON.total_details.amount_shipping,
          "stripe.tax_amount":sessionJSON.total_details.amount_tax
        }
      }
    ]
  }]

  const nrMetricAPI = 'https://metric-api.newrelic.com/metric/v1'
  axios({
    url: nrMetricAPI,
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Api-Key': 'LICENSE_KEY'
     },
    data: JSON.stringify(nrMetrics)
  })
   .then((response) => {
    // Handle the response here if needed
    console.log('Metrics data sent successfully:', response.data);
  })
  .catch((error) => {
    // Handle any errors that occur during the request
    console.error('Error sending metrics data to New Relic:', error);
  })
}

```
**Note:** Change the `sessionJSON` according to your response and call this function where you are getting Stripe metrics in your application.
``````javascript
 sendDataToNewRelic (session)
```

> ## Monitor your application
> You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard:
> 1. Go to **[one.newrelic.com](https://one.newrelic.com/)** and click on **+ Add data**.
> 2.  Click on the **Dashboards** tab.
> 3.  In the search box, type `Stripe`.
> 4.  When you see our pre-build dashboard, click on it to install it in your account.
> 
> Once your application is integrated by following the above steps, the dashboard should display metrics.
> 
> To instrument the Stripe quickstart and to see metrics and alerts, you can also follow our [Stripe quickstart page](https://newrelic.com/instant-observability/stripe) by clicking on the **Install now** button.

Reviewer: ```suggestion
## Monitor your application
You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard, go to [https://onenr.io/0ZQWYYE1rRW] and follow the instructions there. Once installed, the dashboard should display metrics.
```

Thanks @sjyothi54 . I replaced this section with just a single link to go to the UI.

---
title: Stripe integration
tags:
  - Stripe integration
  - New Relic integration
metaDescription: Install our Stripe dashboards and see your Stripe data in New Relic.  
---
import infrastructureStripeDashboard from 'images/infrastructure_screenshot-stripe-dashboard.webp'

By utilizing New Relic's Stripe dashboard, you gain the ability to monitor your application's transaction overview, transaction volume, revenue and transaction errors effortlessly. The Stripe dashboard is developed with our application specific monitoring agent, browser monitoring agent offering you a consolidated view of critical data, conveniently accessible in one central location.

<img
    title="Stripe"
    alt="Stripe dashboard"
    src={infrastructureStripeDashboard}
/>

<figcaption>
After setting up our Stripe, you can install a dashboard for your Stripe metrics. 
</figcaption>

## Install the infrastructure agent [#infra]

> Before getting Stripe data into New Relic, first install our infrastructure agent, then you will be able to get details about your host.
> 
> * Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can install the infrastructure agent by following the docs.
> * By integrating this [infrastructure agent](https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/) in your system, one can able to 
>   know the CPU usage, Memory usage, Storage usage and Network traffic etc.
>  

Reviewer: ```suggestion
Before getting Stripe data into New Relic, first install our infrastructure agent, and then you will be able to get details about your host, such as CPU usage, memory usage, storage usage, network traffic, and more.

Follow our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to instrument your system with the infrastructure agent. Or if you prefer, you can manually install the infrastructure agent by following the [installation docs](/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent/). 
 
```

> ## Install application specific New Relic agent

Reviewer: ```suggestion
## Install an application-specific New Relic agent
```

> * Follow the steps in our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) for a quick and easy install process.

Reviewer: ```suggestion
* Follow the steps in our [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1) to install on of our agents based on your application type.
```

> ## Install Browser monitoring agent

Reviewer: ```suggestion
## Install our browser monitoring agent
```

> Our Browser monitoring agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.

Reviewer: ```suggestion
Our browser monitoring agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.
```

> * To install the agent, Follow this steps to install [Browser Monitoring Agent](https://docs.newrelic.com/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/)

Reviewer: ```suggestion
See [Install the browser monitoring agent](/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/) to get started.
```

> ## Set up synthetic monitoring
> Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Stripe dashboard.
> To set up synthetic monitors, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6)

Reviewer: ```suggestion
## Set up synthetic monitoring
Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Stripe dashboard.

To set up synthetic monitoring, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6).
```

## Send metrics to New Relic

> Check the [Metric API documentation](https://docs.newrelic.com/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.

Reviewer: ```suggestion
Check the [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.
```

> To proceed further, you will need a New Relic license key. Kindly follow the [New Relic API key documentation](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.

Reviewer: ```suggestion
To proceed further, you will need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
```

> On a successful stripe payment, we receive a JSON from the stripe API service.
> We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.
> 
> Below is the javascript code snippet which formats the data and sends it to New Relic metric API URL
> ```

Reviewer: ```suggestion
On a successfulStripe payment, we receive a JSON from the Stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic metric API requirements.

Below is the JavaScript code snippet that formats the data and sends it to New Relic metric API URL:

```
```

> ```

Reviewer: ```suggestion
```javascript
```

          "stripe.amount_total": sessionJSON.amount_total,
          "stripe.customer_email": sessionJSON.customer_details.email,
          "stripe.created_at": sessionJSON.created,
          "stripe.currency" : sessionJSON.currency,
          "stripe.payment_status": sessionJSON.payment_status,
          "stripe.status": sessionJSON.status,
          "stripe.discount_amount":sessionJSON.total_details.amount_discount,
          "stripe.shipping_amount":sessionJSON.total_details.amount_shipping,
          "stripe.tax_amount":sessionJSON.total_details.amount_tax
        }
      }
    ]
  }]

  const nrMetricAPI = 'https://metric-api.newrelic.com/metric/v1'
  axios({
    url: nrMetricAPI,
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Api-Key': 'LICENSE_KEY'
     },
    data: JSON.stringify(nrMetrics)
  })
   .then((response) => {
    // Handle the response here if needed
    console.log('Metrics data sent successfully:', response.data);
  })
  .catch((error) => {
    // Handle any errors that occur during the request
    console.error('Error sending metrics data to New Relic:', error);
  })
}

```

> **Note:** Change the sessionJSON according to your response and call this function where you are getting stripe metrics in your application.

Reviewer: ```suggestion
**Note:** Change the `sessionJSON` according to your response and call this function where you are getting Stripe metrics in your application.
```

> ```

Reviewer: ```suggestion
```javascript
```

 sendDataToNewRelic (session)
```

## Monitor your application

> You can choose our pre-built dashboard template named `Stripe` to monitor your Stripe application metrics.

Reviewer: ```suggestion
You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard:
```

1. Go to **[one.newrelic.com](https://one.newrelic.com/)** and click on **+ Add data**.
2.  Click on the **Dashboards** tab.
3.  In the search box, type `Stripe`.

> 4.  When you see our pre-build dashboard, click on it to install it in your account.

Reviewer: @sjyothi54 I wasn't able to find this dashboard. Can you provide a direct link to it? 
<img width="1332" alt="image" src="https://github.com/newrelic/docs-website/assets/42753584/c9572b8e-b419-4853-8be4-d52890893a65">

Author: Stripe dashboard:https://onenr.io/0ZQWYYE1rRW

Once your application is integrated by following the above steps, the dashboard should display metrics.

> To instrument the Stripe quickstart and to see metrics and alerts, you can also follow our [Stripe quickstart page](https://newrelic.com/instant-observability/stripe) by clicking on the **Install now** button.

Author: @ally-sassman  The quickstart isn't live yet

    </Step>
  <Step>
    ### Install an APM agent [#apm-setup]

Install one of our application performance monitoring (APM) agents in the [guided install](https://one.newrelic.com/nr1-core?state=999174fa-ed8b-d707-af7b-8e4f8fd03cc1).
    </Step>

  <Step>
    ### Install our browser monitoring agent [#browser-setup]

Our browser monitoring agent reports application and browser-related metrics like AJAX performance, error rate, duration, core web vitals, and throughput.
* See [Install the browser monitoring agent](/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/) to get started.
    </Step>

    <Step>
      ### Set up synthetic monitoring [#synthetic-setup]

Our synthetic monitors test your app in different locations and capture server downtime. You can view this data in the Stripe dashboard.

To set up synthetic monitors, follow the steps in our [guided install](https://onenr.io/0EjOXrnxgQ6)
  </Step>

    <Step>

> ### Send metrics to New Relic [#send-metrics]
> 
> Check the [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) for further details.
> 
> To proceed further, you will need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
> On a successful stripe payment, we receive a JSON from the stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.

Reviewer: @sjyothi54 Should this be renamed as an optional step to set up custom metrics? 
```suggestion
### (Optional) Send custom metrics to New Relic [#send-metrics]

If you need to customize the app metrics sent to New Relic, see [Metric API documentation](/docs/data-apis/ingest-apis/metric-api/report-metrics-metric-api/) to learn how. Once you've set up custom metrics, with a successful stripe payment, we receive a JSON from the stripe API service. We need to send these values to New Relic by formatting the JSON as per New Relic Metric API requirements.

If you're sending custom metrics, you'll need a New Relic license key. See [New Relic API Keys](/docs/apis/intro-apis/new-relic-api-keys/) to get your license key.
```

Author: @ally-sassman This step is essential, not optional.

> Below is the javascript code snippet which formats the data and sends it to New Relic metric API URL
> ```javascript

Reviewer: ```suggestion
The following JavaScript code snippet formats the data and sends it to the New Relic metric API URL: 

```javascript
```

          "stripe.status": sessionJSON.status,
          "stripe.discount_amount":sessionJSON.total_details.amount_discount,
          "stripe.shipping_amount":sessionJSON.total_details.amount_shipping,
          "stripe.tax_amount":sessionJSON.total_details.amount_tax
        }
      }
    ]
  }]

  const nrMetricAPI = 'https://metric-api.newrelic.com/metric/v1'
  axios({
    url: nrMetricAPI,
    method: 'POST',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded',
      'Api-Key': 'LICENSE_KEY'
     },
    data: JSON.stringify(nrMetrics)
  })
   .then((response) => {
    // Handle the response here if needed
    console.log('Metrics data sent successfully:', response.data);
  })
  .catch((error) => {
    // Handle any errors that occur during the request
    console.error('Error sending metrics data to New Relic:', error);
  })
}

```

**Note:** Change the `sessionJSON` according to your response and call this function where you are getting Stripe metrics in your application.

```javascript
 sendDataToNewRelic (session)
```

      </Step>

  <Step>
### Install our Stripe monitoring dashboard [#add-dashboard]

> You can choose our pre-built Stripe dashboard to monitor your application metrics. To set up this dashboard, go to [https://onenr.io/0ZQWYYE1rRW] and follow the instructions there. Once installed, the dashboard should display metrics.

Reviewer: ```suggestion
To set up our pre-built Stripe dashboard to monitor your application metrics, go to the [Stripe dashboard installation](https://onenr.io/0ZQWYYE1rRW) and follow the instructions. Once installed, the dashboard should display metrics.
```
